src/components/snackbar/snackbar.tsx

```diff
@@ -99,89 +103,99 @@ export class Snackbar {
     @Event()
     private hide: EventEmitter<void>;
 
-    private mdcSnackbar: MDCSnackbar;
+    @State()
+    private offset: number = 0;
 
-    constructor() {
-        this.handleMdcClosing = this.handleMdcClosing.bind(this);
-    }
+    @State()
+    private open: boolean = false;
 
-    public connectedCallback() {
-        this.initialize();
-    }
+    @State()
+    private closing: boolean = true;
 
-    public componentDidLoad() {
-        this.initialize();
-    }
+    private snackbarId: string;
 
-    private initialize() {
-        const element = this.host.shadowRoot.querySelector('.mdc-snackbar');
-        if (!element) {
-            return;
-        }
-
-        this.mdcSnackbar = new MDCSnackbar(element);
-
-        this.mdcSnackbar.listen('MDCSnackbar:closing', this.handleMdcClosing);
+    public constructor() {
+        this.snackbarId = createRandomString();
     }
 
-    public disconnectedCallback() {
-        this.mdcSnackbar.unlisten('MDCSnackbar:closing', this.handleMdcClosing);
-        this.mdcSnackbar.destroy();
+    @Listen('changeOffset')
+    protected onChangeIndex(event: CustomEvent<number>) {
+        event.stopPropagation();
+        this.offset = event.detail;
     }
 
     /**
      * Show the snackbar
      */
     @Method()
     public async show() {
-        if (this.timeout) {
-            this.mdcSnackbar.timeoutMs = this.timeout;
+        if (this.open) {
+            return;
         }
 
-        this.mdcSnackbar.labelText = this.message;
+        this.open = true;
+        this.closing = false;
+        container.add(this.host);
 
-        this.mdcSnackbar.open();
+        if (this.timeout) {
+            setTimeout(
+                this.close,
+                Math.max(this.timeout - hideAnimationDuration, 0),
+            );
+        }
     }
 
+    private close = () => {
+        if (!this.open) {
+            return false;
+        }
+
+        this.closing = true;
+
+        setTimeout(() => {
+            this.open = false;
+            container.remove(this.host);
+            this.hide.emit();
+            this.offset = 0;
+        }, hideAnimationDuration);
+    };
+
     public render() {
         return (
             <aside
-                class={`
-                    mdc-snackbar
-                    ${this.multiline ? 'mdc-snackbar--stacked' : ''}
-                `}
+                popover="manual"
                 style={{
                     '--snackbar-timeout': `${this.timeout}ms`,
+                    '--snackbar-distance-to-top-edge': `${this.offset}px`,
+                }}
+                class={{
+                    open: this.open,
+                    'is-closing': this.closing,
                 }}
+                id={this.snackbarId}
+                role={this.open ? 'status' : undefined}
+                aria-relevant={this.open ? 'additions' : undefined}
```

Comment:
it seems we have always had a bad implementation of MDC, as these attributes remained static in the DOM in the previous versions.

Information about these accessibility things is generally really confusing when I read about them online. But I guess this is the correct implementation when I check this one, which has some example: https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA22.html

They say:
> This is done by adding `role="status"` to the element that contains the [status message](https://www.w3.org/TR/WCAG21/#dfn-status-messages).

Reply:
Yeah, I've noticed that these things often seem broken. MDC is supposed to set the correct values as you interact with different components, but I'm guessing the problem might be that as you interact with things, we re-render them, and then the values get re-set to what is only intended to be starting values.